### DISCUSSION TITLE
mermaid rendering blank

### DISCUSSION THREAD

**Discussion by andww** [vuepress-theme-hope/vuepress-theme-hope#2372] [Q&A] [answered]
Thank you for a wonderful plugin. I am new to vuepress and am having issues rendering mermaid.

I have configured in my config.js with following:
 plugins: [
    mdEnhancePlugin({
      flowchart: true,
      mermaid: true,
      codetabs: true,
    }),
  ],

In the relevant page, i have md as follows:
```mermaid
flowchart TB
    c1-->a2
    subgraph one
    a1-->a2
    end
    subgraph two
    b1-->b2
    end
    subgraph three
    c1-->c2
    end
    one --> two
    three --> two
    two --> c2
```

```flow
st=>start: Start|past:>http://www.google.com[blank]
e=>end: End|future:>http://www.google.com
op1=>operation: My Operation|past
op2=>operation: Stuff|current
sub1=>subroutine: My Subroutine|invalid
cond=>condition: Yes
or No?|approved:>http://www.google.com
c2=>condition: Good idea|rejected
io=>inputoutput: catch something...|future

st->op1(right)->cond
cond(yes, right)->c2
cond(no)->sub1(left)->op1
c2(yes)->io->e
c2(no)->op2->e
```

The flowchart is rendering well. However, the mermaid sequence diagram is blank.
![Screen Shot 2022-11-05 at 9 42 31 PM](https://user-images.githubusercontent.com/19371060/200122895-ee203333-0c0f-4b4a-b86a-64878869311b.png)

Hope you can give me some light into this. Maybe I am missing some configuration.

Thank you.

**Comment by Mister-Hope** [comment]
I need minimal reproduction.

https://vuepress-theme-hope.github.io/v2/md-enhance/guide/mermaid.html#flowchart

Markdown Enhance plugin docs are enabling all features, and you can see that flowchart works well.

Also the get started page is runing both features

https://vuepress-theme-hope.github.io/v2/md-enhance/guide/#flowchart

**Comment by andww** [comment, marked as answer]
Thanks for the prompt response.

Found out that the mermaid diagram displays after building and deploying on live site. However, it doesn't display in dev in localhost.

**Reply by Mister-Hope** [reply]
That could be a bug, you need to provide a reproduction repo and open an issue

**Reply by Mister-Hope** [reply]
I already fix this in main, the isue is caused by internal change in mermaid 9.2.0. Vite can not handle cjs deps in esm package correctly.
